File: android/app/src/main/java/com/example/jamboree/tutorials/TutorialVoteCard.kt
package com.example.jamboree.tutorials

import androidx.compose.foundation.layout.*
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.ElevatedCard
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/** Snapshot of TutorialVote used for rendering. */
fun TutorialVote.snapshot() = TutorialVoteCardState(
    isOpen = isOpen, yesCount = yesCount, noCount = noCount,
    eligibleCount = eligibleCount, result = result, tutorialShown = tutorialShown,
)

data class TutorialVoteCardState(
    val isOpen: Boolean,
    val yesCount: Int,
    val noCount: Int,
    val eligibleCount: Int,
    val result: Boolean?,
    val tutorialShown: Boolean,
)

/** Shared lobby card mirroring TutorialVoteCard.swift. */
@Composable
fun TutorialVoteCard(
    state: TutorialVoteCardState,
    tutorial: GameTutorial,
    onCall: () -> Unit,
    onVote: (Boolean) -> Unit,
    onDismiss: () -> Unit,
) {
    if (state.tutorialShown) return
    var myVote by remember { mutableStateOf<Boolean?>(null) }

    when {
        state.isOpen -> ElevatedCard {
            Column(Modifier.padding(12.dp)) {
                Text("Show tutorial first?", style = MaterialTheme.typography.titleSmall)
                Text("${state.yesCount + state.noCount} / ${state.eligibleCount} voted — majority wins.",
                     style = MaterialTheme.typography.bodySmall)
                Spacer(Modifier.height(8.dp))
                Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    Button(
                        onClick = { myVote = true; onVote(true) },
                        modifier = Modifier.weight(1f),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = if (myVote == true) Color(0xFF2E7D32) else Color.Gray),
                    ) { Text("Yes (${state.yesCount})") }
                    Button(
                        onClick = { myVote = false; onVote(false) },
                        modifier = Modifier.weight(1f),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = if (myVote == false) Color(0xFFC62828) else Color.Gray),
                    ) { Text("No (${state.noCount})") }
                }
            }
        }
        state.result == true -> ElevatedCard {
            Column(Modifier.padding(12.dp)) {
                var pageIndex by remember { mutableIntStateOf(0) }
                val sections = tutorial.sections
                val total = sections.size
                val section = sections[pageIndex]

                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    Text(tutorial.title, style = MaterialTheme.typography.titleMedium)
                    Text("${pageIndex + 1} / $total",
                         style = MaterialTheme.typography.bodySmall,
                         color = MaterialTheme.colorScheme.onSurfaceVariant)
                }

                Spacer(Modifier.height(8.dp))
                Text(section.heading, style = MaterialTheme.typography.titleSmall)
                Spacer(Modifier.height(2.dp))
                Text(section.body, style = MaterialTheme.typography.bodyMedium)
                Spacer(Modifier.height(12.dp))

                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    OutlinedButton(onClick = onDismiss) { Text("Skip") }
                    Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                        if (pageIndex > 0) {
                            OutlinedButton(onClick = { pageIndex-- }) { Text("← Back") }
                        }
                        if (pageIndex < total - 1) {
                            Button(onClick = { pageIndex++ }) { Text("Next →") }
                        } else {
                            Button(onClick = onDismiss) { Text("Got it — start") }
                        }
                    }
                }
            }
        }
        state.result == false -> Text("Majority voted to skip the tutorial.",
                                       style = MaterialTheme.typography.bodySmall)
        else -> OutlinedButton(onClick = onCall) { Text("Call tutorial vote") }
    }
}

File: ios/Jamboree/Tutorials/TutorialVoteCard.swift
import SwiftUI

/// Shared lobby card: "call tutorial vote" → live tally → result, then a
/// readable summary of the tutorial sections with a "Got it" dismiss button
/// once the vote passes. Used by every browser-tier host view.
struct TutorialVoteCard: View {
    /// Mirrors `TutorialVote` snapshot fields we need for rendering.
    struct VoteState {
        var isOpen: Bool
        var yesCount: Int
        var noCount: Int
        var eligibleCount: Int
        var result: Bool?
        var tutorialShown: Bool
    }

    let state: VoteState
    let tutorial: GameTutorial
    let onCall: () -> Void
    let onVote: (Bool) -> Void
    let onDismiss: () -> Void

    @State private var myVote: Bool?
    @State private var pageIndex = 0

    init(state: VoteState, tutorial: GameTutorial, onCall: @escaping () -> Void, onVote: @escaping (Bool) -> Void, onDismiss: @escaping () -> Void) {
        self.state = state
        self.tutorial = tutorial
        self.onCall = onCall
        self.onVote = onVote
        self.onDismiss = onDismiss
    }

    var body: some View {
        if state.tutorialShown {
            EmptyView()
        } else if state.isOpen {
            GroupBox("Show tutorial first?") {
                Text("\(state.yesCount + state.noCount) / \(state.eligibleCount) voted — majority wins.")
                    .font(.caption).foregroundStyle(.secondary)
                HStack {
                    Button { myVote = true; onVote(true) } label: {
                        Text("Yes (\(state.yesCount))").frame(maxWidth: .infinity)
                    }
                    .buttonStyle(.borderedProminent)
                    .tint(myVote == true ? .green : .gray)

                    Button { myVote = false; onVote(false) } label: {
                        Text("No (\(state.noCount))").frame(maxWidth: .infinity)
                    }
                    .buttonStyle(.borderedProminent)
                    .tint(myVote == false ? .red : .gray)
                }
            }
        } else if state.result == true {
            GroupBox(tutorial.title) {
                let sections = tutorial.sections
                let total = sections.count
                let section = sections[pageIndex]

                Text("\(pageIndex + 1) / \(total)")
                    .font(.caption).foregroundStyle(.secondary)
                    .frame(maxWidth: .infinity, alignment: .trailing)

                VStack(alignment: .leading, spacing: 4) {
                    Text(section.heading).font(.subheadline.bold())
                    Text(section.body).font(.callout).foregroundStyle(.secondary)
                }
                .padding(.vertical, 8)

                HStack {
                    Button("Skip") { onDismiss() }
                        .buttonStyle(.bordered)
                        .tint(.secondary)
                    Spacer()
                    if pageIndex > 0 {
                        Button("← Back") { pageIndex -= 1 }.buttonStyle(.bordered)
                    }
                    if pageIndex < total - 1 {
                        Button("Next →") { pageIndex += 1 }.buttonStyle(.borderedProminent)
                    } else {
                        Button("Got it — start") { onDismiss() }.buttonStyle(.borderedProminent)
                    }
                }
            }
        } else if state.result == false {
            Text("Majority voted to skip the tutorial.")
                .foregroundStyle(.secondary)
        } else {
            Button("Call tutorial vote", action: onCall)
                .buttonStyle(.bordered)
        }
    }
}
